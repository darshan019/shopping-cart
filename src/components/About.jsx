import React from "react";
import styles from "./css/about.module.css"

export default function About() {
  return (
    <div className={styles.about}>
      <h2>This is a fictional store and none of the products displayed here exist.</h2>
      <h3>Products' information and images: Fake Store API.</h3>
      <h3>darshan019: <a target="_blank" href="https://github.com/darshan019">Github</a></h3>
    </div>
  )
}