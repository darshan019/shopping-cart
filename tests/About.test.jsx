import About from "../src/components/About"
import { render } from "@testing-library/react"

describe("About Component", () => {
  it("renders the about component correctly", () => {
    const { container, getByText, getByRole } = render(<About />);
    expect(container).toMatchSnapshot()

    const mainHeading = getByText("This is a fictional store and none of the products displayed here exist.");
    expect(mainHeading).toBeInTheDocument();

    const paragraphs = container.querySelectorAll("p");
    expect(paragraphs.length).toBe(2); 

    
    const githubLink = getByText("darshan019:");
    expect(githubLink).toBeInTheDocument();
    
    
    const linkElement = getByRole('link', { name: /Github/i });
    expect(linkElement).toHaveAttribute("href", "https://github.com/darshan019");
    expect(linkElement).toHaveAttribute("target", "_blank");
  });
});