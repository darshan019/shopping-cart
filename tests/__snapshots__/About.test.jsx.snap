// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Component > renders the about component correctly 1`] = `
<div>
  <div
    class="_about_698222"
  >
    <h1>
      This is a fictional store and none of the products displayed here exist.
    </h1>
    <p>
      Products' information and images: Fake Store API.
    </p>
    <p>
      darshan019: 
      <a
        href="https://github.com/darshan019"
        target="_blank"
      >
        Github
      </a>
    </p>
  </div>
</div>
`;
